vmware/build-tools-for-vmware-aria
# [artifact-manager] [archetypes/vrops] (#280) Add support for priority ordering of Aria Operations policies

### Description

Aria Operations (vROPs) since version 8.17.0 supports ordering policies by priority. If the target server has Aria Operations version 8.17 and later the policies will be ordered by priority during pushing of policies to the server as they appear in the `content.yaml` file with the first policy with highest priority and the last one with lowest priority, hence making possible to control the priority of the policies in the configuration file. If the Aria Operation server version is older no ordering by priority will be done during pushing of policies (in order to maintain backwards compatibility).

Example content.yaml file with policies ordered by priority.

```yaml
policy:
- "Policy 1"
- "Policy 2"
- "Policy 3"
```

If the Aria Operations server is 8.17.0 and later during push the policies will be ordered by priority as follows:

1. Policy 1 (top priority)
2. Policy 2
3. Policy 3 (lowest priority)

Note that the Aria Operations server is 8.17.0 does not support the default policy to be part of the prioritization, hence it will be excluded from the prioritization list during push (but it will be imported).

### Checklist

- [x] I have added relevant error handling and logging messages to help troubleshooting
- [x] I have updated CHANGELOG.md with a short summary of the changes introduced

### Testing

1. Verify that the Aria Operation (vROPs) is version 8.17.x.
2. Add vROPs policies in `content.yaml` file ordering them by priority.
3. Do `mvn clean package vrealize:push -P<env>` for a vROPs project.
5. Verify the policy priority in the vROPs UI.

### Release Notes

* [vrops] Issue 280 / Support for priority ordering of the vROPs policies during push.

### Related issues and PRs

Closed #280

### Target System UI

![Screenshot_280_Policy_Priority](https://github.com/vmware/build-tools-for-vmware-aria/assets/125263901/aea41d77-6993-458d-93d1-4ef7eebf5df2)
